### PR TITLE
Make internal ClusterInfo::getPlan() wait for initial plan load from agency

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Otherwise the maintenance, fired off by the heartbeat thread, may not wait for the initial data to arrive from the agency, which can cause Plan and Current to be empty when maintenance starts.
No backport for 3.6 is needed, as 3.6 has blocking agency communication here in case the Plan is not there.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] Backports required for: 3.7 (https://github.com/arangodb/arangodb/pull/13291)

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as *mini chaos*.

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13367/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13370/

**Review by goedderz — COMMENTED** [review]
I think it'd be a very good idea to additionally make `waitForPlan` and `waitForCurrent` `[[nodiscard]]` to prevent something like this in the future.

File: CHANGELOG [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+
 * Reduce overhead of audit logging functionality if audit logging is turned
   off.
 
 * Add several more attributes to audit-logged queries, namely query execution
   time and exit code (0 = no error, other values correspond to general ArangoDB 
   error codes).
 
+* Make internal ClusterInfo::getPlan() wait for initial plan load from agency.
+
```
Comment (goedderz) [inline CHANGELOG:14]:
```suggestion
```
The entry made it here twice.

File: arangod/Cluster/DBServerAgencySync.cpp [hunk 2]
```diff
@@ -204,6 +204,17 @@ DBServerAgencySyncResult DBServerAgencySync::execute() {
...
   AgencyCache::databases_t plan = clusterInfo.getPlan(planIndex, dirty);
 
+  if (plan.empty()) {
+    // this early exit is a safeguard against overzealous cleanup by 
+    // the maintenance. in case the plan is empty and compared to the
+    // local state, the maintenance may decide to drop everyting 💥
+    LOG_TOPIC("0a6f2", DEBUG, Logger::MAINTENANCE)
+        << "DBServerAgencySync::execute no plan change set";
+    result.success = true;
+    result.errorMessage = "DBServerAgencySync::execute no reconciliation plan";
```
Comment (goedderz) [inline arangod/Cluster/DBServerAgencySync.cpp:214]: Shouldn't that be
```suggestion
    result.success = false;
    result.errorMessage = "DBServerAgencySync::execute no reconciliation plan";
```
instead?

Comment (goedderz) [reply arangod/Cluster/DBServerAgencySync.cpp:214]: Or is it considered successful in the sense, that it has done everything it should on the plan that's available?

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13374/

**Comment by jsteemann** [comment]
Tests blue

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13382/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13395/

**Review by fceller — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13462/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13465/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**